src/crm/contacts.service.ts

```
@@ -1673,8 +1674,43 @@ export class ContactsService {
   async remove(id: number): Promise<void> {
-    await this.repository.delete(id);
+    const tenantId = this.tenantContext.requireTenant();
+    const contact = await this.repository.findOne({
+      where: { id, tenant: { id: tenantId } },
+    });
+
+    if (!contact) {
+      throw new NotFoundException(`Contact with id ${id} not found`);
+    }
+
+    await this.connection.transaction(async (manager) => {
```

ESLint

[error] Unexpected space before function parentheses.

(space-before-function-paren)